main/vstring.h

```diff
@@ -30,7 +30,7 @@
 #endif
 #ifdef VSTRING_PUTC_MACRO
 #define vStringPut(s,c) \
-	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : 0), \
+	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : (void) 0 ), \
 	((s)->buffer [(s)->length] = (c)), \
 	((c) == '\0' ? 0 : ((s)->buffer [++(s)->length] = '\0')))
 #endif
```

Comment:
@codebrainz, thank you.

@t-b, your macro is better than mine. Could you convert it to an inline functionas @codebrainz wrote. Then add the change as a commit to this PR; your original change and converting the original macro to the inline function should be in separated commits.